### PR TITLE
feat: implement app routing

### LINKED ISSUE #40: Implement Routing

### PR DESCRIPTION
**Pull request by NikaKlokava**
**Description:**

1. Create components:
   - AboutComponent
   - DashboardComponent
      - ProjectsComponent;
   - AuthComponent:
      - LoginComponent;
      - SignupComponent;
   - ProfileComponent;
   - NotFoundComponent;
  
 2. Implement routing;
 3. Add router-outlet for each child route;
 4. Fix components prefix (from "**app-**" to "**nb-**");
 5. Fix tests (10/10);
 6. Create Shared folder with subfolders (pipes, directives, components, guards, services).

### CODE REVIEW

File: angular.json
Comment (Howl404) [inline angular.json:1]: Change prefix "app" to "nb"

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+      },
+    ],
+  },
+  {
+    path: 'dashboard',
+    loadComponent: () =>
+      import('./pages/dashboard/dashboard.component').then(
+        (m) => m.DashboardComponent
+      ),
+    children: [
```
Comment (marustas) [inline src/app/app.routes.ts]: This shouldn't be a child of `dashboard` route, but a sibling

**Review by marustas — CHANGES_REQUESTED** [review]

**Comment by Howl404** [comment]
Also please use PR template from any other PR

**Review by IrinaIID — COMMENTED** [review]
Please change folder structure: move profile to Dashboard (it is a part of that)

File: src/app/app.routes.ts (hunk 1 above)
Comment (NikaKlokava) [reply src/app/app.routes.ts]: you mean, it should be an independent route?:
{
path: 'dashboard/projects',

loadComponent: () =>
          import('./pages/dashboard/projects/projects.component').then(
            (m) => m.ProjectsComponent
          ),
}

**Pull request closed without merging**